File: typecast-csharp/src/Typecast/Models/AgeEnum.cs
using System.Text.Json.Serialization;

namespace Typecast.Models;

/// <summary>
/// Age classification for voice filtering.
/// </summary>
[JsonConverter(typeof(JsonStringEnumMemberConverter))]
public enum AgeEnum
{
    /// <summary>Child voice (approximately 0-12 years)</summary>
    [JsonPropertyName("child")] Child,
    
    /// <summary>Teenager voice (approximately 13-19 years)</summary>
    [JsonPropertyName("teenager")] Teenager,
    
    /// <summary>Young adult voice (approximately 20-35 years)</summary>
    [JsonPropertyName("young_adult")] YoungAdult,
    
    /// <summary>Middle-aged voice (approximately 36-55 years)</summary>
    [JsonPropertyName("middle_age")] MiddleAge,
    
    /// <summary>Elder voice (approximately 56+ years)</summary>
    [JsonPropertyName("elder")] Elder
}

/// <summary>
/// Extension methods for AgeEnum.
/// </summary>
public static class AgeEnumExtensions
{
    /// <summary>
    /// Converts the AgeEnum to its API string representation.
    /// </summary>
    public static string ToApiString(this AgeEnum age) => age switch
    {
        AgeEnum.Child => "child",
        AgeEnum.Teenager => "teenager",
        AgeEnum.YoungAdult => "young_adult",
        AgeEnum.MiddleAge => "middle_age",
        AgeEnum.Elder => "elder",
        _ => throw new ArgumentOutOfRangeException(nameof(age))
    };

    /// <summary>
    /// Parses a string to AgeEnum.
    /// </summary>
    public static AgeEnum ParseAge(string value) => value.ToLowerInvariant() switch
    {
        "child" => AgeEnum.Child,
        "teenager" => AgeEnum.Teenager,
        "young_adult" => AgeEnum.YoungAdult,
        "middle_age" => AgeEnum.MiddleAge,
        "elder" => AgeEnum.Elder,
        _ => throw new ArgumentException($"Unknown age: {value}", nameof(value))
    };
}

File: typecast-java/src/main/java/com/neosapience/models/AgeEnum.java
package com.neosapience.models;

/**
 * Age category enumeration for voice filtering.
 */
public enum AgeEnum {
    /** Child voice */
    CHILD("child"),
    /** Teenager voice */
    TEENAGER("teenager"),
    /** Young adult voice */
    YOUNG_ADULT("young_adult"),
    /** Middle age voice */
    MIDDLE_AGE("middle_age"),
    /** Elder voice */
    ELDER("elder");

    private final String value;

    AgeEnum(String value) {
        this.value = value;
    }

    /**
     * Returns the string value of the age category.
     * 
     * @return the age category string
     */
    public String getValue() {
        return value;
    }

    /**
     * Creates an AgeEnum from a string value.
     * 
     * @param value the age category string
     * @return the corresponding AgeEnum
     * @throws IllegalArgumentException if the value doesn't match any age category
     */
    public static AgeEnum fromValue(String value) {
        if (value == null) {
            return null;
        }
        for (AgeEnum age : values()) {
            if (age.value.equalsIgnoreCase(value)) {
                return age;
            }
        }
        throw new IllegalArgumentException("Unknown age category: " + value);
    }

    @Override
    public String toString() {
        return value;
    }
}
